heroic-shell/src/main/java/com/spotify/heroic/HeroicShell.java

```diff
@@ -175,21 +178,31 @@ public void shutdown() throws Exception {
         });
     }
 
-    static void interactive(Parameters params, CoreInterface core) {
+    static void interactive(Parameters params, CoreInterface core) throws Exception {
         log.info("Setting up interactive shell...");
 
+        Exception e = null;
         try {
             runInteractiveShell(core);
-        } catch (Exception e) {
+        } catch (final Exception inner) {
             log.error("Error when running shell", e);
```

Comment:
Logs the wrong exception, also, if you log it in the catch above there is no need to log here.